Source/WTF/wtf/Vector.h

```diff
@@ -108,217 +94,113 @@ struct VectorInitializer<true, false, T>
     template<typename... Args>
     static void initializeWithArgs(T* begin, T* end, Args&&... args)
     {
+        static_assert(VectorTraits<T>::needsInitialization);
+        static_assert(!VectorTraits<T>::canInitializeWithMemset);
```

Comment:
Why add these assertions? This seems an unnecessary new restriction. Maybe a caller chooses to initialize something with arguments that doesn't need initialization or that can be initialized with `memset`? Why wouldn't we support that?

Reply:
I created a test case that demonstrates the problem of the original code. WebKit doesn't have code for that case because it doesn't need it yet.
https://gist.github.com/fujii/7c2d49f351dcf26100211d7e595c1b48